lib/src/sha384_512.dart

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'digest.dart';
+import 'hash.dart';
+import 'hash_sink.dart';
+import 'utils.dart';
+
+/// An instance of [Sha2Sha384].
+///
+/// This instance provides convenient access to the [Sha384][rfc] hash function.
+///
+/// [rfc]: http://tools.ietf.org/html/rfc6234
+final sha384 = Sha384._();
+
+/// An instance of [Sha2Sha512].
+///
+/// This instance provides convenient access to the [Sha512][rfc] hash function.
+///
+/// [rfc]: http://tools.ietf.org/html/rfc6234
+final sha512 = Sha512._();
+
+/// An implementation of the [SHA-384][rfc] hash function.
+///
+/// [rfc]: http://tools.ietf.org/html/rfc6234
+///
+/// Note that it's almost always easier to use [sha384] rather than creating a
+/// new instance.
+class Sha384 extends Hash {
+  @override
+  final int blockSize = 16 * bytesPerWord;
```

Comment:
This looks wrong to me. According to https://en.wikipedia.org/wiki/SHA-2, SHA512 has 1024 bits internal block size, i.e. 128 bytes. `16 * 4` is 64.

Reply:
On this note, that should have failed HMAC-SHA512 test, had there been one.

Reply:
You're commenting on an outdated commit. Also, this is not HMAC-512, its just Sha512.

Reply:
I see what you're getting at though - I did not update the HMAC. That can be easily done (pull requests welcome ;) )

Reply:
#71